src/app/modules/training-event/services/event.service.ts

```diff
@@ -1,21 +1,56 @@
-import { Injectable } from '@angular/core';
+import { Injectable, Inject} from '@angular/core';
+import { timer, of as observableOf } from 'rxjs';
+import { catchError, takeWhile } from 'rxjs/operators';
 import { HttpClient } from '@angular/common/http';
 
 import { environment } from 'src/environments/environment';
 import {
+  CampusEventResponse,
   OffCampusEventRequest,
   OffCampusEventResponse,
 } from 'src/app/interfaces/event';
+import { AUTH_SERVICE, AuthService } from 'src/app/interfaces/auth-service';
+import { PaginatedResponse } from 'src/app/interfaces/paginated-response';
 
 /** Provide services for Event. */
 @Injectable({
   providedIn: 'root'
 })
 export class EventService {
+ /** How often should we query latest notifications. */
+ private REFRESH_INTERVAL = 30 * 1000;
```

Comment:
I doubt that this auto-reload feature is required in your case. Normally, we don't need the list to be refreshed if no input is performed by user.